113.md

```diff
@@ -0,0 +1,141 @@
+
+In general, activities are notes that have an activity file associated with them (gpx, tcx, fit), and a summary can be produced about this activity.
+
+The structure of an activity follows very closely with [NIP-23](23.md) long-form content events.
+### Content
+The `.content` field MAY contain a description of the activity
+### Activity Types
+Activities MUST include a `t` tag with an activity type (defined in Appendix A). e.g. `run | walk | hike`
+### Activity File
+Activities MAY include an `r` tag, which will include a URL to the location where an activity file can be found.
+### Metadata
+`t` - The activity type  
+`r` - location of activity file (OPTIONAL)  
+`title` - The title of the activity  
+`recorded_at` - The unix timestamp (in seconds) when the activity was recorded  
+`encrypted` - An indication that this event is encrypted (OPTIONAL)
+
+```json
+{
+  "kind": 30100,
+  "content": "<description-of-activity>",
+  "tags": [
+	 ["title", "Title of Activity"],
+	 ["t", "<activity-type-from-nip>"]
+	 ["d", "UUID generated by client"],
+	 ["r", "url to download activity file"],
+	 ["published_at", "<unix-timestamp>"],
+	 ["updated_at", "<unix-timestamp>"],
+	 ["recorded_at", "<unix-timestamp>"],
+	 ["encrypted"] // Optional, activity file is encrypted
+  ],
+}
+```
+### Summaries
+`kind:30101` defines an event which is meant to summarize activities, users, places, etc...
+Summaries are intended to capture a wide range things that users may wish to summarize in relation to activities. 
+Users will likely want a summary of a specific event (e.g. milage, time, pace, etc...), a summary for a group of events (e.g. miles run this month), a summary of locations (e.g. heatmaps based on a geohash). 
+
+This NIP does not attempt to describe all the different summaries that users could want, as this list will be ever evolving. Rather this event type should facilitate clients to retrieve relevant summaries for a user.
+#### Summary Metadata
+`r` - An ID that this summary pertains to. It could be an activity ID or a user pubkey  
+`t` - A summary MUST include a type which clients can specifically look for. These types must be lowercase with no spaces. e.g. `sports-lib-summary`  
+`g` - Optional geohash of a summary  
+`encrypted` - Optional, if summary is encrypted.  
+`content` - stringified payload of the summary - the structure of which is defined by the type.  
+`unit` - Optional, defines the unit types that are present in a summary. Either `imperial | metric`.
+
+```json
+{
+  "kind": 30101,
+  "content": "JSON.stringify(summary)", 
+  "tags": [
+	 ["d", "UUID generated by client"],
+	 ["r", "<UUID-of-activity>"],
+	 ["t", "general-summary"],
+	 ["g", "OPTIONAL geohash"],
+	 ["unit", "OPTIONAL imperial | metric"],
+	 ["published_at", "<unix-timestamp>"],
+	 ["updated_at", "<unix-timestamp>"],
+	 ["encrypted"] // Optional, summary content is encrypted
```

Comment:
is this necessary? I feel like encrypted is not necessary and just complicates things. if you feel like encrypted is important, make it a different kind so clients can opt out of fetching those if they want to display all public activities and save everybody's bandwidth.

Reply:
Yeah perhaps a different kind for encrypted things might be better. The main dilemma as I see it is a lot of this data is quite personal (gpx files run the risk of doxxing your exact locations), so I wanted the ability for clients to still publish summaries so they're synced across devices, but without revealing the contents of the summary.

I different kind could work as well, and definitely will make it easier for clients to filter them out.

Reply:
so clients could publish both events, each of a kind, one encrypted for personal use and one for public consumption. when fetching the user personal history you just fetch encrypted stuff, when fetching stuff from others you fetch public.